src/jit/valuenum.cpp

```diff
@@ -4982,15 +4983,17 @@ void Compiler::fgValueNumberBlockAssignment(GenTreePtr tree, bool evalAsgLhsInd)
                         assert(lhs->OperGet() == GT_IND);
                         lhsAddr = lhs->gtOp.gtOp1;
                     }
+
                     // For addr-of-local expressions, lib/cons shouldn't matter.
                     assert(lhsAddr->gtVNPair.BothEqual());
                     ValueNum lhsAddrVN = lhsAddr->GetVN(VNK_Liberal);
 
                     // Unpack the PtrToLoc value number of the address.
                     assert(vnStore->IsVNFunc(lhsAddrVN));
+
                     VNFuncApp lhsAddrFuncApp;
                     vnStore->GetVNFunc(lhsAddrVN, &lhsAddrFuncApp);
-                    assert(lhsAddrFuncApp.m_func == VNF_PtrToLoc);
+
```

Comment:
Ah, whoops. I meant to restore that assertion--good catch. I am seeing `VNF_PtrToLoc`, yes.